src/main/main.ts

```diff
@@ -148,6 +159,45 @@ ipcMain.on(ChannelEnum.HEALTH_CHECK, async (event) => {
   }
 });
 
+ipcMain.on(ChannelEnum.LOAD_PRESET, async (event, preset_name: string) => {
+  const channel = ChannelEnum.LOAD_PRESET;
+  console.log(`Loading preset: ${preset_name}`);
+
+  // TODO: should we do some str checking here?
```

Comment:
While we aren't doing strict str checking, we're throwing an error if we fail to fetch so that seems good enough to me. @xenown what do you think about this?

Reply:
I think its fine. 177 and 178 won't throw exceptions.